doc/migration/version-upgrade-guide.md

```diff
@@ -0,0 +1,177 @@
+For some versions it might also be necessary to update the upstream Kubernetes and controller-runtime dependencies.
+
+The full list of dependencies and their versions required by a particular version of the SDK can be viewed by using the [`operator-sdk print-deps`][print-deps-cli]. Use this command to update the project `Gopkg.toml`/`go.mod` file accordingly as you upgrade to a new SDK version.
+
+
+For some SDK versions after `v0.1.0` there might be minor breaking changes in the controller-runtime APIs, `operator-sdk` CLI or the expected project layout and file names. These breaking changes will usually be outlined in the [CHANGELOG][changelog]/[release-notes][release-notes] for each release version.
+
+For releases that have a large number of breaking changes or involve a significant refactoring of the APIs and project layout there will be a migration guide similar to the [v0.1.0 migration guide][v0.1.0-migration-guide].
+
+The following sections outline the upgrade steps for each SDK version along with steps necessary for any associated breaking changes.
+
+## `v0.2.x`
+
+Update the SDK constraint in `Gopkg.toml` to use version `v0.2.1` and run `dep ensure` to update the vendor.
+```TOML
+[[constraint]]
+  name = "github.com/operator-framework/operator-sdk"
+  version = "=v0.2.1"
+```
+
+## `v0.3.x`
+
+Update the SDK constraint in `Gopkg.toml` to use version `v0.3.0`, the kubernetes dependencies to `kubernetes-1.12.3` revisions, and the controller-runtime version to `v0.1.8`. Then run `dep ensure` to update the vendor.
+```TOML
+[[override]]
+  name = "k8s.io/code-generator"
+  # revision for tag "kubernetes-1.12.3"
+  revision = "3dcf91f64f638563e5106f21f50c31fa361c918d"
+
+[[override]]
+  name = "k8s.io/api"
+  # revision for tag "kubernetes-1.12.3"
+  revision = "b503174bad5991eb66f18247f52e41c3258f6348"
+
+[[override]]
+  name = "k8s.io/apiextensions-apiserver"
+  # revision for tag "kubernetes-1.12.3"
+  revision = "0cd23ebeb6882bd1cdc2cb15fc7b2d72e8a86a5b"
+
+[[override]]
+  name = "k8s.io/apimachinery"
+  # revision for tag "kubernetes-1.12.3"
+  revision = "eddba98df674a16931d2d4ba75edc3a389bf633a"
+
+[[override]]
+  name = "k8s.io/client-go"
+  # revision for tag "kubernetes-1.12.3"
+  revision = "d082d5923d3cc0bfbb066ee5fbdea3d0ca79acf8"
+
+[[override]]
+  name = "sigs.k8s.io/controller-runtime"
+  version = "=v0.1.8"
+
+[[constraint]]
+  name = "github.com/operator-framework/operator-sdk"
+  version = "=v0.3.0"
+```
+
+## `v0.4.x`
+
+Update the SDK constraint in `Gopkg.toml` to use version `v0.4.1` and run `dep ensure` to update the vendor.
+```TOML
+[[constraint]]
+  name = "github.com/operator-framework/operator-sdk"
+  version = "=v0.4.1"
+```
+
+## `v0.5.x`
+
+- Update the SDK constraint in `Gopkg.toml` to use version `v0.5.0`, the kubernetes dependencies to `kubernetes-1.13.1` revisions, and the controller-runtime version to `v0.1.10`.
```

Comment:
We should either use bullets everywhere or nowhere IMO. Makes the doc easier to read.